app/src/main/java/com/sopt/dive/signup/SignUpActivity.kt

```diff
@@ -0,0 +1,261 @@
+                finish()
+            }
+        }
+    }
+}
+
+// 2. 회원가입 페이지
+@Composable
+fun SignUpScreen(
+    onSignUpClick: (String, String, String, String) -> Unit
+) {
+    // ID와 PW를 각각 별도의 상태로 관리
+    var idText by remember { mutableStateOf("") }
+    var pwText by remember { mutableStateOf("") }
+    var nicknameText by remember { mutableStateOf("") }
+    var numberText by remember { mutableStateOf("") }
+
+    Column( // Q :이거 컴포넌트 효율적으로 쓰는 방법없음 ? 참조로 바든 방법이라든가 이거 코드너무커진다
+        modifier = Modifier
+            .fillMaxSize() // 화면 전체를 채움
+            .padding(horizontal = 24.dp) // 좌우 여백 24dp 적용
+    ) {
+        // 상단 타이틀
+        Text(
+            text = "SIGN UP",
+            modifier = Modifier
+                .fillMaxWidth() // 가로 전체를 채움
+                .padding(top = 50.dp, bottom = 40.dp), // 상하 여백 적용
+            textAlign = TextAlign.Center, // 텍스트 중앙 정렬
+            fontSize = 30.sp,
+            fontWeight = FontWeight.Bold,
+        )
+
+        // ID 입력
+        Column(modifier = Modifier) {
+            Text(text = "ID", fontSize = 20.sp)
+
+            BasicTextField(
+                value = idText,
+                onValueChange = { idText = it },
+                modifier = Modifier.fillMaxWidth(),
+                decorationBox = { innerTextField ->
+                    Box(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .background(
+                                color = Color(0xFFE0E0E0),
+                                shape = RoundedCornerShape(size = 16.dp)
+                            )
+                            .padding(horizontal = 16.dp, vertical = 16.dp)
+                    ) {
+                        if (idText.isEmpty()) {
+                            Text(
+                                text = "아이디를 입력해 주세요.",
+                                fontSize = 18.sp,
+                                color = Color.Gray,
+                            )
+                        }
+                        innerTextField()
+                    }
+                }
+            )
+        }
+
+        Spacer(modifier = Modifier.height(24.dp))
+
+        // PW 입력
+        Column(modifier = Modifier) {
+            Text(text = "PW", fontSize = 20.sp)
+
+            BasicTextField(
+                value = pwText,
+                onValueChange = { pwText = it },
+                modifier = Modifier.fillMaxWidth(),
+                visualTransformation = PasswordVisualTransformation(),
+                decorationBox = { innerTextField ->
+                    Box(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .background(
+                                color = Color(0xFFE0E0E0),
+                                shape = RoundedCornerShape(size = 16.dp)
+                            )
+                            .padding(horizontal = 16.dp, vertical = 16.dp)
```

Comment:
```suggestion
                            .fillMaxWidth()
                            .background(
                                color = Color(0xFFE0E0E0),
                                shape = RoundedCornerShape(size = 16.dp)
                            )
                            .padding(vertical = 16.dp)
```
저의 경우에는 컴포넌트에 `fillMaxWidth`를 둔 경우 vertical 패딩만 두는 편입니다ㅎㅎ